{
    "body": "I'm having problem with a package that fails to install properly, but still claims to have done so.  See below for relevant part of the log file.\n\n(NOTE: This unrecognized error was obtained by adding \"--enable-shared\" to the configure() command in package.py.)\n\nIs there a Spack option that keeps the stage around for inspection even after a \"successful\" install?\n\n-- Elizabeth\n\n```\ncfei_hypre.cxx:(.text._ZN3MPI2Op4InitEPFvPKvPviRKNS_8DatatypeEEb[_ZN3MPI2Op4InitEPFvPKvPviRKNS_8DatatypeEEb]+0x18): undefined reference to `ompi_mpi_cxx_op_intercept'\n/tmp/rpfische/spack-stage/spack-stage-u5iMsT/hypre-2.10.0b/src/FEI_mv/fei-hypre/cfei_hypre.o: In function `MPI::Intracomm::Create_graph(int, int const*, int const*, bool) const':\ncfei_hypre.cxx:(.text._ZNK3MPI9Intracomm12Create_graphEiPKiS2_b[_ZNK3MPI9Intracomm12Create_graphEiPKiS2_b]+0x28): undefined reference to `MPI::Comm::Comm()'\n/tmp/rpfische/spack-stage/spack-stage-u5iMsT/hypre-2.10.0b/src/FEI_mv/fei-hypre/cfei_hypre.o: In function `MPI::Graphcomm::Clone() const':\ncfei_hypre.cxx:(.text._ZNK3MPI9Graphcomm5CloneEv[_ZNK3MPI9Graphcomm5CloneEv]+0x25): undefined reference to `MPI::Comm::Comm()'\n/tmp/rpfische/spack-stage/spack-stage-u5iMsT/hypre-2.10.0b/src/FEI_mv/fei-hypre/cfei_hypre.o: In function `MPI::Cartcomm::Clone() const':\ncfei_hypre.cxx:(.text._ZNK3MPI8Cartcomm5CloneEv[_ZNK3MPI8Cartcomm5CloneEv]+0x25): undefined reference to `MPI::Comm::Comm()'\n/tmp/rpfische/spack-stage/spack-stage-u5iMsT/hypre-2.10.0b/src/FEI_mv/fei-hypre/cfei_hypre.o: In function `MPI::Intercomm::Merge(bool) const':\ncfei_hypre.cxx:(.text._ZNK3MPI9Intercomm5MergeEb[_ZNK3MPI9Intercomm5MergeEb]+0x27): undefined reference to `MPI::Comm::Comm()'\n/tmp/rpfische/spack-stage/spack-stage-u5iMsT/hypre-2.10.0b/src/FEI_mv/fei-hypre/cfei_hypre.o: In function `MPI::Intracomm::Create(MPI::Group const&) const':\ncfei_hypre.cxx:(.text._ZNK3MPI9Intracomm6CreateERKNS_5GroupE[_ZNK3MPI9Intracomm6CreateERKNS_5GroupE]+0x2a): undefined reference to `MPI::Comm::Comm()'\n/tmp/rpfische/spack-stage/spack-stage-u5iMsT/hypre-2.10.0b/src/FEI_mv/fei-hypre/cfei_hypre.o:cfei_hypre.cxx:(.text._ZNK3MPI9Intracomm5SplitEii[_ZNK3MPI9Intracomm5SplitEii]+0x26): more undefined references to `MPI::Comm::Comm()' follow\n/tmp/rpfische/spack-stage/spack-stage-u5iMsT/hypre-2.10.0b/src/FEI_mv/fei-hypre/cfei_hypre.o:(.data.rel.ro._ZTVN3MPI8DatatypeE[_ZTVN3MPI8DatatypeE]+0x78): undefined reference to `MPI::Datatype::Free()'\n/tmp/rpfische/spack-stage/spack-stage-u5iMsT/hypre-2.10.0b/src/FEI_mv/fei-hypre/cfei_hypre.o:(.data.rel.ro._ZTVN3MPI3WinE[_ZTVN3MPI3WinE]+0x48): undefined reference to `MPI::Win::Free()'\ncollect2: error: ld returned 1 exit status\nmake[1]: *** [libHYPRE.so] Error 1\nmake[1]: Leaving directory `/tmp/rpfische/spack-stage/spack-stage-u5iMsT/hypre-2.10.0b/src/lib'\n\nInstalling hypre ...\nmkdir -p -- /home/rpfische/spack/opt/spack/linux-x86_64/gcc-4.9.3/hypre-2.10.0b-d4ed63tt5dncdjplfv6jjz55xevbkj3y/lib /home/rpfische/spack/opt/spack/linux-x86_64/gcc-4.9.3/hypre-2.10.0b-d4ed63tt5dncdjplfv6jjz55xevbkj3y/include\ncp: cannot stat \u2018/tmp/rpfische/spack-stage/spack-stage-u5iMsT/hypre-2.10.0b/src/hypre/lib/*\u2019: No such file or directory\n\n==> Successfully installed hypre\n  Fetch: 6.61s.  Build: 40.01s.  Total: 46.62s\n[+] /home/rpfische/spack/opt/spack/linux-x86_64/gcc-4.9.3/hypre-2.10.0b-d4ed63tt5dncdjplfv6jjz55xevbkj3y\n[rpfische@ankeli build]$ \n```\n",
    "user": "citibeth",
    "url": "https://api.github.com/repos/spack/spack/issues/505",
    "updated_at": "2016-03-16 15:38:47",
    "created_at": "2016-03-08 16:34:52",
    "closed_at": "2016-03-16 15:38:47",
    "state": "closed",
    "title": "hypre/package.py has error on install, but claims to install successfully",
    "number": 505,
    "milestone": null,
    "labels": [],
    "id": 139325221,
    "html_url": "https://github.com/spack/spack/issues/505",
    "assignees": [],
    "comments": 2
}